### PR TITLE
feat(SwipeAction): add `onActionsReveal` prop

### PR DESCRIPTION
**Pull request by LadyChatterleyLover**
feat(swipeAction): add onActionsReveal props

### CODE REVIEW

**Review by zqran — CHANGES_REQUESTED** [review]

**Comment by zqran** [comment]
https://github.com/ant-design/ant-design-mobile/blob/bafad206d6cdbacbd45ea699918ddda8cfa5e326/src/components/swipe-action/swipe-action.tsx#L141-L142

指令式展示 action 操作按钮时，也应该触发 `onActionsReveal`

**Comment by LadyChatterleyLover** [comment]
@zqran 已经添加了哈

**Comment by zqran** [comment]
image
上面的文档内容也修改下吧~

**Comment by LadyChatterleyLover** [comment]
@zqran 文档已经修改了哈，

**Review by awmleer — APPROVED** [review]

File: src/components/swipe-action/swipe-action.tsx [hunk 1]
```diff
@@ -141,10 +146,12 @@ export const SwipeAction = forwardRef<SwipeActionRef, SwipeActionProps>(
     useImperativeHandle(ref, () => ({
       show: (side: 'left' | 'right' = 'right') => {
         if (side === 'right') {
+          p.onActionsReveal?.('right')
```
Comment (zqran) [inline src/components/swipe-action/swipe-action.tsx:149]: 这块的逻辑，可以这样处理不

```tsx
useImperativeHandle(ref, () => ({
  show: (side: 'left' | 'right' = 'right') => {
    if (side === 'right') {
      api.start({
        x: -getRightWidth(),
      })
    } else if (side === 'left') {
      api.start({
        x: getLeftWidth(),
      })
    }
    // 这边，提到外面
    p.onActionsReveal?.(side)
  },
  close,
}))
```

Comment (awmleer) [reply src/components/swipe-action/swipe-action.tsx:149]: @LadyChatterleyLover 你最近有空调整一下这里不？

**Comment by LadyChatterleyLover** [comment]
好的，晚上调整一下，



发自我的iPhone


------------------ Original ------------------
From: zqran ***@***.***&gt;
Date: Thu,Jul 21,2022 5:56 PM
To: ant-design/ant-design-mobile ***@***.***&gt;
Cc: luopei ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [ant-design/ant-design-mobile] feat(SwipeAction): add`onActionsReveal` prop (PR #5484)





 
@zqran commented on this pull request.
 
 
In src/components/swipe-action/swipe-action.tsx:
 &gt; @@ -141,10 +146,12 @@ export const SwipeAction = forwardRef<SwipeActionRef, SwipeActionProps&gt;(      useImperativeHandle(ref, () =&gt; ({        show: (side: 'left' | 'right' = 'right') =&gt; {          if (side === 'right') { +          p.onActionsReveal?.('right')  
这块的逻辑，可以这样处理不
 useImperativeHandle(ref, () =&gt; ({   show: (side: 'left' | 'right' = 'right') =&gt; {     if (side === 'right') {       api.start({         x: -getRightWidth(),       })     } else if (side === 'left') {       api.start({         x: getLeftWidth(),       })     }     // 这边，提到外面     p.onActionsReveal?.(side)   },   close, }))

**Comment by LadyChatterleyLover** [comment]
好的，晚上调整一下，



发自我的iPhone


------------------ Original ------------------
From: 卡晨 ***@***.***&gt;
Date: Thu,Jul 21,2022 6:09 PM
To: ant-design/ant-design-mobile ***@***.***&gt;
Cc: luopei ***@***.***&gt;, Mention ***@***.***&gt;
Subject: Re: [ant-design/ant-design-mobile] feat(SwipeAction): add`onActionsReveal` prop (PR #5484)





 
@awmleer commented on this pull request.
 
 
In src/components/swipe-action/swipe-action.tsx:
 &gt; @@ -141,10 +146,12 @@ export const SwipeAction = forwardRef<SwipeActionRef, SwipeActionProps&gt;(      useImperativeHandle(ref, () =&gt; ({        show: (side: 'left' | 'right' = 'right') =&gt; {          if (side === 'right') { +          p.onActionsReveal?.('right')  
@LadyChatterleyLover 你最近有空调整一下这里不？

**Comment by LadyChatterleyLover** [comment]
@zqran @awmleer 已经调整了哈，

**Review by awmleer — APPROVED** [review]
LGTM

**Pull request merged**